rollout/trafficrouting.go

```diff
@@ -233,10 +234,10 @@ func (c *rolloutContext) reconcileTrafficRouting() error {
 				desiredWeight = 100
 			}
 		}
-
-		// We need to check for Generation > 1 because when we first install the rollout we run step 0 this prevents that.
-		// We could also probably use c.newRS == nil || c.newRS.Status.AvailableReplicas == 0
-		if currentStep != nil && c.rollout.ObjectMeta.Generation > 1 {
+		// We need to check for revision > 1 because when we first install the rollout we run step 0 this prevents that.
+		// There is a bigger fix needed for the reasons on why we run step 0 on rollout install, that needs to be explored.
+		revision, revisionFound := annotations.GetRevisionAnnotation(c.rollout)
+		if currentStep != nil && (revisionFound && revision > 1) {
```

Comment:
@huoyinghao Would you mind double checking that this also fixes your use case, I think this is a better fix after having spent time looking into it?

Reply:
i also think so, thanks